### PR TITLE
Support multi-line directives

### LINKED ISSUE #163: Trailing commas not supported for per-file-ignores value
When used in a `.pylintrc` file, the `per-file-ignores` key does not support a value with a trailing comma even though trailing commas are valid for pylintrc-flavored INI.

### Example

This configuration should be valid but isn't:
```
per-file-ignores=
    **/*_foo.py:missing-module-docstring,
```

One currently needs to remove the trailing comma for the configuration to parse successfully:
```
per-file-ignores=
    **/*_foo.py:missing-module-docstring
```

If one does not remove the trailing comma, the Pylint server emits this error:
> pylint.exceptions.UnknownMessageError: No such message id or symbol ''.

Presumably because the parser expects a Pylint code after the comma.

### Versions

- `pylint` 4.0.5
- `python` 3.14.2
- `pylint-per-file-ignores` 3.2.0

### Additional context

Based on the examples in the README, this seems to impact configuration in `setup.cfg` as well. It also seems like it might be working as intended? Filing this bug anyway to confirm. If it is WAI, could you consider documenting in the README that 1) trailing commas are not supported and 2) a newline is the only valid delimiter between patterns?

### PR DESCRIPTION
**Pull request by ndmeiri**
* Updates `_plugin.py` to support trailing commas in directives.
* Adds tests to exercise more formatting styles.
  * The `ConfigTestCase` dataclass encapsulates a single test case that generalizes to both INI and TOML formats.
  * The `ConfigTestCase.Descriptor` dataclass encodes the formatting style under test.

Closes #163.

### CODE REVIEW

File: test/test_config_parsing.py [hunk 1]
```diff
@@ -61,33 +63,462 @@ def test_without_per_file_ignores(runner: Runner) -> None:
     assert _get_symbols(result, "b") == ["invalid-name", "missing-module-docstring"]
 
 
-def test_config_pyproject_toml(runner: Runner) -> None:
-    """Test per-file-ignores config parsed from pyproject.toml."""
-    result = runner("test_config_pyproject_toml")
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class ConfigTestCase:
```
Comment (kasium) [inline test/test_config_parsing.py:67]: I get the point writing tests in this style but please write code similar to existing ones. I would accept a PR with two tests that show that directives spanning over multiple lines work

**Review by kasium — CHANGES_REQUESTED** [review]

File: test/test_config_parsing.py (hunk 1 above)
Comment (ndmeiri) [reply test/test_config_parsing.py:67]: Thanks for the review. Is the objection related to the number of tests, the style of the tests, or that some of the tests are arguably unrelated to the implementation change in this PR?

Here’s some context around why these tests ended up written in this style:

- The existing test coverage made it difficult to change the implementation to add trailing comma support while being confident that I wouldn’t break parsing of other valid formats.
- So to ensure I didn’t break existing functionality, I initially added tests in the existing test style, but that was _very_ unruly and unreadable due to the number of duplicated boilerplate lines in each test function.
  - (I also tried writing a single parametrized test with four Boolean parameters—mirroring the parameters in `ConfigTestCase`—and dynamically generating the `ini_content` based on the values of those parameters, but that was even less unreadable. Dynamically generating config file content is not a scalable approach.)
- So to minimize boilerplate in each test, I extracted the salient parts of each test case into `ConfigTestCase`.

If the objection is more about the number of tests, then I’d be happy to discuss why this level of coverage is important and improves maintainability.

If the objection is more about having arguably unrelated tests in this PR, then I’d be happy to move them to a subsequent PR.

Comment (kasium) [reply test/test_config_parsing.py:67]: First, I explained which tests I would expect. I don't see any critical level of duplication here.
Second, I'm not willing to discuss this with an AI. Don't get me wrong, using AI is fine but if you see that a comment was written by a human, I don't want to see an AI generate reply. Especially given the following context

- I have explained which tests I would expect
- The level of complexity and maintainability was my initial concern. I don't need an AI explaining me that a 450(!) lines of tests are needed to safeguard such a feature

Comment (ndmeiri) [reply test/test_config_parsing.py:67]: Hey thanks for the reply. That comment was genuinely written 100% by me, a human. AI never touched it, proofread it, or generated any part of it. (Rhetorical question: Did the em-dashes contribute to the perception? 😄 Ask any of my coworkers and they'll tell you that I've been using those since way before generative AI came along.)

> First, I explained which tests I would expect.

I actually found the initial comment ambiguous which is why I tried to clarify in good faith and explain my rationale so we can discuss further as two software engineers trying to reach consensus. To be honest, I'm not sure why my effort to volunteer a contribution to this project is being met with what feels from my perspective like antagonism at best and hostility at worst. Would appreciate if we could communicate in good faith in both directions.

My starting position is that we're working _together_ to find the best solution to a problem while closing the gap between any differences of opinion through respectful discussion. Ideally, that discussion would be data driven and centered on engineering principles.

> Second, I'm not willing to discuss this with an AI.

Don't worry; you're not :) I'm a human with a penchant for verbosity. Especially when trying to communicate with a stranger in writing on the Internet, I err on the side of over communicating to be explicit and avoid ambiguity. It has served me well during my FAANG tenure and I try to lean into that approach when I communicate in other relevant spaces.

> 450(!) lines of tests

Agreed the size of the PR is larger than should be needed. But I'd posit that the large addition of tests in this PR is merely a symptom of a lack of tests at HEAD. If most of these test cases were already covered at HEAD, then the PR would be much smaller because I would have been able to make the change with higher confidence that I wasn't breaking existing functionality.

> to safeguard such a feature

If these test cases were already covered at HEAD, then we would've caught the regression in #165, which intended to implement this exact feature. So yes, these tests not only help safeguard this feature but also help protect the correctness of the parsing code for nearly all other config formats in both INI and TOML files. Shouldn't we consider that a net win?

Sure, new developers will pay a one-time cost of a few minutes spent understanding how the tests are structured. But after that, adding tests and making changes will be simpler and less error prone, thanks in part to less boilerplate. Overall adding these tests seems like 1) best practice and 2) a significant win for maintainability and both short-term and long-term project health.

> The level of complexity and maintainability was my initial concern.

Gotcha. I've tried to respond in good faith to this concern in my two comments. More than happy to discuss further if you'd like to respond to the rationale that I shared. Perhaps there's a way to decrease complexity and improve maintainability while still increasing test coverage. Open to suggestions.

> I have explained which tests I would expect

Again, I'm genuinely still unsure which tests you'd like to see. "two tests that show that directives spanning over multiple lines work" leaves quite a bit of room for interpretation. I'm not sure which test cases you'd like to see. There are several relevant combinations of config formatting parameters, many of which are likely to be used in the wild.

Also, if the plan is to add just two tests, how can we be sure that the code under test behaves correctly for the other combinations of parameters?

Comment (kasium) [reply test/test_config_parsing.py:67]: Okay, first things first: I did not want to frame you as an AI :-) Also I don't mind if people use AI coding agents to write code. The only things was, that you comment has a certain style and phrases which are normally used by LLM models (but maybe you just have a very nice writing style). So again, sorry for that.
Second, thanks a lot for the contribution and the time you invest. This is highly appreciated.
Third, the tests were not there when you opened the issue. Therefore there was no test catching the error. Before you opened this PR, I added the tests to prevent further regressions.

Now let's talk about what we love - code. I still think that this is too boilerplate code. In addition, the tests are written in a style which I and my colleagues don't use. This is not a criticism - it's a personal and subjective decision. Yet, as I need to maintain it within the next years, I would like to keep things in the same shape. So how can we adapt your change? Quite easy - split test test data and the setup/assertion logic. Test files can be moved to testdirs, while the logic remains in the test code. Then the dataclasses could be transformed into single test functions or pytest.mark.parameterize. 

I also took now the change to write a test and I came up with the following:

```py
@pytest.mark.parametrize("trailing_comma", [True, False])
def test_config_pylintrc_multiline(runner: Runner, *, trailing_comma: bool) -> None:
    """Test per-file-ignores config parsed from .pylintrc with multiline settings."""
    cwd = runner.datadir / "test_config_pylintrc_multiline"
    if trailing_comma:
        pylintrc = cwd / ".pylintrc"
        content = pylintrc.read_text()
        content = content.replace("unused-import", "unused-import,")
        pylintrc.write_text(content)

    result = runner("test_config_pylintrc_multiline")

    assert _get_symbols(result, "a") == ["invalid-name"]
    assert _get_symbols(result, "b") == ["missing-module-docstring"]
```

And the config-file `test/test_config_parsing/test_config_pylintrc_multiline/.pylintrc`:

```
[MAIN]
load-plugins = pylint_per_file_ignores

[MESSAGES CONTROL]
per-file-ignores =
    a.py:missing-module-docstring,
        unused-import
    b.py:C0103
```

What do you think about this? IMO it's shorter and has a significantly smaller diff. However, I admit that your tests maybe catch more corner cases but I'm not sure if the costs are worth it.

**Comment by kasium** [comment]
Stale PR - let me implement this as I think it's a useful addítion. Anyways thanks for your work
